packages/config/src/loadConfigFromFile.ts

```diff
@@ -145,28 +134,92 @@ export async function loadConfigFromFile({
   }
 }
 
-async function requireTypeScriptFile(resolvedPath: string) {
+async function loadConfigTsOrJs(configRoot: string, configFile: string | undefined) {
+  const { loadConfig: loadConfigWithC12 } = await import('c12')
+  const { deepmerge } = await import('deepmerge-ts')
+
   try {
-    const jiti = createJiti(__filename, {
-      interopDefault: true,
-      moduleCache: false,
+    const {
+      config,
+      configFile: resolvedPath,
+      meta,
+    } = await loadConfigWithC12({
+      cwd: configRoot,
+      // configuration base name
+      name: 'prisma',
+      // the config file to load (without file extensions), defaulting to `${cwd}.${name}`
+      configFile,
+      // do not load .env files
+      dotenv: false,
+      // do not load RC config
+      rcFile: false,
+      // do not extend remote config files
+      giget: false,
+      // do not extend the default config
+      extend: false,
+      // do not load from nearest package.json
+      packageJson: false,
+      // specify the default config to use
+      // defaultConfig: defaultConfig(),
+
+      // @ts-expect-error: this is a type-error in `c12` itself
+      merger: deepmerge,
+
+      jitiOptions: {
+        interopDefault: true,
+        moduleCache: true,
```

Comment:
This change was unintended.

```suggestion
        moduleCache: false,
```